maturity-level-two/src/Extensions/ApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using AutoMapper;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Http;
+using Swashbuckle.AspNetCore.Swagger;
+using Swashbuckle.AspNetCore.SwaggerUI;
+
+namespace Codit.LevelTwo.Extensions
+{
+    public static class ApplicationBuilderExtensions
+    {
+        /// <summary>
+        ///     Configure to use AutoMapper
+        /// </summary>
+        /// <param name="applicationbuilder">Application builder to use</param>
+        /// <param name="mapperConfigExpression">AutoMapper configuration to use</param>
+        public static void UseAutoMapper(this IApplicationBuilder applicationbuilder, Action<IMapperConfigurationExpression> mapperConfigExpression)
+        {
+            Mapper.Initialize(mapperConfigExpression);
+        }
+
+        /// <summary>
+        ///     Configure to use global exception handler with application/problem+json
+        /// </summary>
+        /// <param name="applicationBuilder">Application builder to use</param>
+        public static void UseExceptionHandlerWithProblemJson(this IApplicationBuilder applicationBuilder)
+        {
+            applicationBuilder?.UseExceptionHandler(errorApplication =>
+            {
+                errorApplication.Run(async context =>
+                {
+                    var errorFeature = context.Features.Get<IExceptionHandlerFeature>();
+                    var exception = errorFeature.Error;
+
+                    var errorDetail = context.Request.IsLocalRequest()
+                        ? exception.Demystify().ToString()
+                        : "The instance value should be used to identify the problem when calling customer support";
+
+                    var problemDetails = new ProblemDetailsError
+                    {
+                        Title = "An unexpected error occurred!",
+                        Status = StatusCodes.Status500InternalServerError,
+                        Detail = errorDetail,
+                        Instance = $"urn:codit.eu:server-error:{Guid.NewGuid()}"
+                    };
+
+                    // TODO: Plug in telemetry
+                    context.Response.WriteJson(problemDetails, contentType: ContentTypeNames.Application.JsonProblem);
```

Comment:
Shouldn't we let ASP.NET handle the serialization of the error in the format that is specified by the accept header of the request instead of deciding for ourselves that it should always be json ?

Reply:
@fgheysels  adding this comment to the Content-Negotiation issue